# Historical DNA Methylation Analysis: CpG O/E
Author: Emma Strand
Last edited: 20200527

Background on CpG O/E: [Dimond et al 2016](https://onlinelibrary.wiley.com/doi/pdf/10.1111/mec.13414?casa_token=J5-ThKmLEYsAAAAA:y5x-yut6DwxurmYAVWRa-xDoybiwM-9pzEloQ3gbADjz6zc5v4HPaNALEK29LropHhwIWjBmNANRQg), [Dixon et al 2014](https://bmcgenomics.biomedcentral.com/track/pdf/10.1186/1471-2164-15-1109), and [Putnam et al 2016](https://onlinelibrary.wiley.com/doi/pdf/10.1111/eva.12408).

*Montipora capitata* gene sets downloaded from [Nucleotide CDS link here](http://cyanophora.rutgers.edu/montipora/) (Shumaker et al 2018). This will download as "Mcap.mRNA.fa.gz".




*Pocillopora acuta* genome downloaded from [here](http://ihpe.univ-perp.fr/acces-aux-donnees/) (Vidal-Dupiol et al 2019). This will download as "Data_to_download.rar" which requires an application to open ([how to open .rar on Mac](https://www.businessinsider.com/how-to-open-rar-on-mac)). Once opened, the file needed is "Pocillopora_acuta_genome_v1.fasta".   

**Calculating CpG O/E**  
Based on scripts and pipeline instructions from [Dimond et al 2016](https://onlinelibrary.wiley.com/doi/pdf/10.1111/mec.13414?casa_token=J5-ThKmLEYsAAAAA:y5x-yut6DwxurmYAVWRa-xDoybiwM-9pzEloQ3gbADjz6zc5v4HPaNALEK29LropHhwIWjBmNANRQg).  

1. Navigate to the desired directory where .fa/.fasta files are located.  

```
$ cd MyProjects/CpG_OE
```

2. Unzip the *Montipora capitata* fasta file.  

```
$ gunzip Mcap.mRNA.fa.gz
```

3. Checking format, header, and number of seqs in *M. cap* file.

```
$ head -5 Mcap.mRNA.fa

output:
>adi2mcaRNA10013_R0
ATGCAGCAGGGAGGAGTCATAATGTACGATGGCAAAACAAGGCGACCTAGCCGAAACGACGACCTCCACACCAAAATGCA
AAGATCTGGTTCTGAAACCCTTTTGAAGACAGACCTTGAACAGTCAAAGGAGGCCATAAAAATCGAAATTCGCAAACAAC
TCAAGCTGAAAGAAGGGGCCGAAAATTTAAAGAAAGTTGCCACGGACAAGAAAACTTTAGCGCAATGCAACTCGATTTTA
AAAGAGGCCACCGCTAAACTTCAAGATCTTCACGATGAATTGCAAGATCTTAATGCAAGAGTGTCGGAGGATGTGCCTTC

$ fgrep -c ">" Mcap.mRNA.fa
output: 63227
```

4. Converting FASTA to tabular format and placing output in the designated directory.

```
$ perl -e '$count=0; $len=0; while(<>) {s/\r?\n//; s/\t/ /g; if (s/^>//) { if ($. != 1) {print "\n"} s/ |$/\t/; $count++; $_ .= "\t";} else {s/ //g; $len += length($_)} print $_;} print "\n"; warn "\nConverted $count FASTA records in $. lines to tabular format\nTotal sequence length: $len\n\n";' \ Mcap.mRNA.fa > fasta2tab

output:
Converted 63227 FASTA records in 1012582 lines to tabular format
Total sequence length: 73464001
```

5. Checking the header in the new fasta2tab file.

```
$ head -2 fasta2tab
output:
adi2mcaRNA10013_R0		ATGCAGCAGGGAGGAGTCATAATGTACGATGGCAAAACAAGGCGACCTAGCCGAAACGACGACCTCCACACCAAAATGCAAAGATCTGGTTCTGAAACCCTTTTGAAGACAGACCTTGAACAGTCAAAGGAGGCCATAAAAATCGAAATTCGCAAACAACTCAAGCTGAAAGAAGGGGCCGAAAATTTAAAGAAAGTTGCCACGGACAAGAAAACTTTAGCGCAATGCAACTCGATTTTAAAAGAGGCCACCGCTAAACTTCAAGATCTTCACGATGAATTGCAAGATCTTAATGCAAGAGTGTCGGAGGATGTGCCTTCTGAAAATGATGATGATACACTTTCGCCAGACAAAACAGATGGTGTGTCCAATGGCCCTGGTAGTAGTAGAATAGAATCCTTAAAAAAGCAAATTGCTGTTGAAATGAAAGTAAAGCAGGGTGCAGAAAACATGTTGGCAATGTACATTGCCAGTGGATCCAAGAAAAAGGACAGTAAACTTACGGCAGAAGCTCAGCAAATGTTGGAGGACTCGAGAAGAAAGATTGAAATTATTCGAATGGCGTTACTGCGGGAACAACAACAAGGGATGCCGGTGGGAAGGGAAGATGGTGTTTTGATGGGTGGCTCTAGTGGAAAAGAAAATGATGCTCCTGGATTAGGACTTTCTCCTTTGGAATTAAGAGTAGAGGATGTGCGACATCACATTGAAATTGAATCTAGGGTAATACAGGGTGCAAAAAATATGATAAAGTTTTTATCCAAGTCTGGTCAAGATAAAGGTTTGGTGGAGGAGGCTGAAAGAAGATTCCATGAATCAACTCAAAAACTAGACTTACTTCAATGCTCACTTACAAGACATATTGGAGAACTATCTCCTGAGCAAGCAGCTCTAGGAATTTACAAAGACTATAACACACTACCTTCATATGCTGATCGGAAGTCTGTTGGTTTGGGGAAACAGCAATACACCGCTATAACTAAACCAGCAGCGCTCACTGGAGAATTAGACGTATGCATCACTGGTGCAGACAGACTGTTAGCAGTTGTTCCCAATCGAGATCCTAGAAAAACTCTCAGTTTGGCAATGGTAAGTCCAGATGGCAAGATCCAAGGAGAGCGCACAGGATCCAGCAAGAGGAAAAGCGGATATAATAAATGGAGTCTGGATGAAAGTTTATCAGTTGAAGTGTGTGCAGTCCTAAAGCTGGATAATAGCTACGTTGGACAAACCTCGTGGAAGGCATTAGGGACGGAGTGTTGGAAGGAGCAGTTTCATATTGAGTTAGACAAGTCTCGAGAACTAGAGGTGGGAGTTTATTGGCGGGATCATAGATCACTTTGTGGAGTTTCATTTTTAAGACTGGAAGAGTTTCTCGATAATCAGCGGCATCAAATTCAAGTCCCTCTTGAACCCCAAGGCTGCCTTTTTGCTGAGGTAACATTTCTTAACCCCATGGTGACAAAGAAACCAAGGCTTCAGCGACAGAAGAAGTTGTTTCATATTCAGAAAGGTGAGAATTTAGGAAAGAATTTTCTTCGGGCTGGTCAGATGAACACAAACGTTGCAACATGGGCCAGATTACTGAAGAGAGCGGCGCCACCGAACTGTACAACCAATGACTCTACTCTTAGTCCAAATTCAACATCACATCCCTTTCCGAGTTCATCACAACGGCCTCAAACACAGTCACCCCCGAGACAACCCGACTTCAACAGCACCACGCCTGGCAAGAACACGAAACCATCAAACAACACTGGGGTACCAGTGGGGTCCACACCAAAAAAATCAGTGAACGAAAATGAAGTCCAGGATGCTCTTTCAGCTTTTGGATTCTTAGATAGCCCAAGTCCTCAAAGATCTGATCTGCAGCCCCCTCCTCAAAAACCAGCTCCACCTGTTAGACGAAAACCTCCCTCACCTCCCACAAGTATAATCAGTCACATGAACATTGACAGTTTTCGATGCATCAGTGTACTGGGCAGAGGCCATTTTGGAAAGGTTATTTTAGCTGAGTACAGGAACACTAATGAACTGTTTGCCATCAAGGCCTTAAAAAAGGGAGACATCATTTCACGAGAAGAAGTTGAAAGTTTGTTGTCTGAGAAGAGAATATTCTTGACAGCAAACAAGATGCGACACCCATTCTTAGTCAACTTATTTGCATGTTTTCAAACTGAGGAGGAACATGTGTGTTTTGTAATGGAGTATGCCCCTGGTGGAGATCTCATGATGCACATCCATGCAGATGTTTTCTCAGAACCTAGAACAGTGTTTTACACGTCATGTGTTGTGTTAGGCCTTCAATATCTCCATGAACACGAAATAGTTTACAGAGATTTGAAGCTGGACAATTTACTGCTGGATTCTGAAGGCTACGTGAAAATTGCTGACTTCGGTTTGTGTAAAGAAGGGATGGGTTATGGAGCACGAACAGGCACCTTCTGTGGCACTCCCGAGTTTTTAGCCCCAGAGGTTCTTACAGAGACCTCGTATACTCGTGCTGTCGACTGGTGGGGACTCGGTGTTCTTATCTTTGAAATGCTTGTAGGAGAGTCTCCGTTTCCGGGAGATGACGAGGAAGAGGTTTTTGACTCTATTGTAAATGACGAAGTTCGTTACCCAAGATTTCTTTCAACGGAAGCTATTGCGATAATGAGAAGGTTACTCAGACGGAACCCCGAGAGGAGGTTAGGAGCCACCGAAAAAGATGCAGAGGATGTTAGAAAACAGCCTTTCTTTAGGGACATGAATTGGGACGACCTTCTAGCAAGAAAAGTTAGACCTCCTTTTGTCCCGAGTGTGGTAAGTACTACTCAATTGCTTATTATTTCAGTTGGGGAATTTTTTTGTCATTGGTTTATGGCTCAAACTTATTCCCCATGA
adi2mcaRNA10016_R5		TTATTCCAAAATATCCTCTTAACAACAGCAAGACAGCATGTTGAGTTTTCGTGTTCAACCGCCGGTGATGGTATTTATCGCAATCCCGCCGACTGTTCCAAGTTCATAGAGTGTCATGGTGGACGTGCATTCCACAAAAGCTGTTCAGCGAAGCTTTTGTTTAACCCTACAATCAAGGCCTGCTCGTTCTCTTGTTTATCTGTTGGTGACGGCATTTATCCAGATCCCGCTGATTGCTCCAAGTTTGTTTTCTGTCATGGCAGTCGTATATTCCTGCACAGCTGTCCACCAGGGCTCCTGTTCAACCCAATTCTCAAAGTTTGCGATTGGCCACAAAATGTTCTGTATGCATTAATATAA
```

6. Add column with length of sequence

```
$ perl -e '$col = 2;' -e 'while (<>) { s/\r?\n//; @F = split /\t/, $_; $len = length($F[$col]); print "$_\t$len\n" } warn "\nAdded column with length of column $col for $. lines.\n\n";' \
fasta2tab > tab_1

output:
Added column with length of column 2 for 63227 lines.
```

7. Checking number of lines in tab file.

```
$ wc tab_1

output:
63227  189681 74412989 tab_1
```

8. Creating file that will be used to count Cs and Gs.

```
$ awk '{print $2}' tab_1 > tab_2
```

9. Counting CGs.

```
$ echo "CG" | awk -F\[Cc][Gg] '{print NF-1}' tab_2 > CG
$ awk '{s+=$1}END{print s}' CG

Output: 2429784
```

10. Counts Cs.

```
$ echo "C" | awk -F\[Cc] '{print NF-1}' tab_2 > C
$ awk '{s+=$1}END{print s}' C

Output: 15227145
```

11. Counts Gs.

```
$ echo "G" | awk -F\[Gg] '{print NF-1}' tab_2 > G
$ awk '{s+=$1}END{print s}' G

Output: 17090960
```

12. Combining the above counts into a new file called comb.

```
$ paste tab_1 \
CG \
C \
G \
> comb
```

13. Viewing the first and last line of the comb file.

```
$ head -1 comb
output:

adi2mcaRNA10013_R0		ATGCAGCAGGGAGGAGTCATAATGTACGATGGCAAAACAAGGCGACCTAGCCGAAACGACGACCTCCACACCAAAATGCAAAGATCTGGTTCTGAAACCCTTTTGAAGACAGACCTTGAACAGTCAAAGGAGGCCATAAAAATCGAAATTCGCAAACAACTCAAGCTGAAAGAAGGGGCCGAAAATTTAAAGAAAGTTGCCACGGACAAGAAAACTTTAGCGCAATGCAACTCGATTTTAAAAGAGGCCACCGCTAAACTTCAAGATCTTCACGATGAATTGCAAGATCTTAATGCAAGAGTGTCGGAGGATGTGCCTTCTGAAAATGATGATGATACACTTTCGCCAGACAAAACAGATGGTGTGTCCAATGGCCCTGGTAGTAGTAGAATAGAATCCTTAAAAAAGCAAATTGCTGTTGAAATGAAAGTAAAGCAGGGTGCAGAAAACATGTTGGCAATGTACATTGCCAGTGGATCCAAGAAAAAGGACAGTAAACTTACGGCAGAAGCTCAGCAAATGTTGGAGGACTCGAGAAGAAAGATTGAAATTATTCGAATGGCGTTACTGCGGGAACAACAACAAGGGATGCCGGTGGGAAGGGAAGATGGTGTTTTGATGGGTGGCTCTAGTGGAAAAGAAAATGATGCTCCTGGATTAGGACTTTCTCCTTTGGAATTAAGAGTAGAGGATGTGCGACATCACATTGAAATTGAATCTAGGGTAATACAGGGTGCAAAAAATATGATAAAGTTTTTATCCAAGTCTGGTCAAGATAAAGGTTTGGTGGAGGAGGCTGAAAGAAGATTCCATGAATCAACTCAAAAACTAGACTTACTTCAATGCTCACTTACAAGACATATTGGAGAACTATCTCCTGAGCAAGCAGCTCTAGGAATTTACAAAGACTATAACACACTACCTTCATATGCTGATCGGAAGTCTGTTGGTTTGGGGAAACAGCAATACACCGCTATAACTAAACCAGCAGCGCTCACTGGAGAATTAGACGTATGCATCACTGGTGCAGACAGACTGTTAGCAGTTGTTCCCAATCGAGATCCTAGAAAAACTCTCAGTTTGGCAATGGTAAGTCCAGATGGCAAGATCCAAGGAGAGCGCACAGGATCCAGCAAGAGGAAAAGCGGATATAATAAATGGAGTCTGGATGAAAGTTTATCAGTTGAAGTGTGTGCAGTCCTAAAGCTGGATAATAGCTACGTTGGACAAACCTCGTGGAAGGCATTAGGGACGGAGTGTTGGAAGGAGCAGTTTCATATTGAGTTAGACAAGTCTCGAGAACTAGAGGTGGGAGTTTATTGGCGGGATCATAGATCACTTTGTGGAGTTTCATTTTTAAGACTGGAAGAGTTTCTCGATAATCAGCGGCATCAAATTCAAGTCCCTCTTGAACCCCAAGGCTGCCTTTTTGCTGAGGTAACATTTCTTAACCCCATGGTGACAAAGAAACCAAGGCTTCAGCGACAGAAGAAGTTGTTTCATATTCAGAAAGGTGAGAATTTAGGAAAGAATTTTCTTCGGGCTGGTCAGATGAACACAAACGTTGCAACATGGGCCAGATTACTGAAGAGAGCGGCGCCACCGAACTGTACAACCAATGACTCTACTCTTAGTCCAAATTCAACATCACATCCCTTTCCGAGTTCATCACAACGGCCTCAAACACAGTCACCCCCGAGACAACCCGACTTCAACAGCACCACGCCTGGCAAGAACACGAAACCATCAAACAACACTGGGGTACCAGTGGGGTCCACACCAAAAAAATCAGTGAACGAAAATGAAGTCCAGGATGCTCTTTCAGCTTTTGGATTCTTAGATAGCCCAAGTCCTCAAAGATCTGATCTGCAGCCCCCTCCTCAAAAACCAGCTCCACCTGTTAGACGAAAACCTCCCTCACCTCCCACAAGTATAATCAGTCACATGAACATTGACAGTTTTCGATGCATCAGTGTACTGGGCAGAGGCCATTTTGGAAAGGTTATTTTAGCTGAGTACAGGAACACTAATGAACTGTTTGCCATCAAGGCCTTAAAAAAGGGAGACATCATTTCACGAGAAGAAGTTGAAAGTTTGTTGTCTGAGAAGAGAATATTCTTGACAGCAAACAAGATGCGACACCCATTCTTAGTCAACTTATTTGCATGTTTTCAAACTGAGGAGGAACATGTGTGTTTTGTAATGGAGTATGCCCCTGGTGGAGATCTCATGATGCACATCCATGCAGATGTTTTCTCAGAACCTAGAACAGTGTTTTACACGTCATGTGTTGTGTTAGGCCTTCAATATCTCCATGAACACGAAATAGTTTACAGAGATTTGAAGCTGGACAATTTACTGCTGGATTCTGAAGGCTACGTGAAAATTGCTGACTTCGGTTTGTGTAAAGAAGGGATGGGTTATGGAGCACGAACAGGCACCTTCTGTGGCACTCCCGAGTTTTTAGCCCCAGAGGTTCTTACAGAGACCTCGTATACTCGTGCTGTCGACTGGTGGGGACTCGGTGTTCTTATCTTTGAAATGCTTGTAGGAGAGTCTCCGTTTCCGGGAGATGACGAGGAAGAGGTTTTTGACTCTATTGTAAATGACGAAGTTCGTTACCCAAGATTTCTTTCAACGGAAGCTATTGCGATAATGAGAAGGTTACTCAGACGGAACCCCGAGAGGAGGTTAGGAGCCACCGAAAAAGATGCAGAGGATGTTAGAAAACAGCCTTTCTTTAGGGACATGAATTGGGACGACCTTCTAGCAAGAAAAGTTAGACCTCCTTTTGTCCCGAGTGTGGTAAGTACTACTCAATTGCTTATTATTTCAGTTGGGGAATTTTTTTGTCATTGGTTTATGGCTCAAACTTATTCCCCATGA	2886	75	569	685

$ tail -1 comb
output:

g9999		ATGGTAAGAGTTGACCATGGCACCGAGGCTTGCCTGATGCTTTTTGTACAGAACCTATTGAGAAACGAAAGGGGGAACCTGAGTTGTGAACCATATATCCAGACACAGTCTAAGCAGAATTTGCCTGCAGAACGCAAGTGGCCAGAAGTCAATCAACGAGTCATTTACCCGATAAAAGATGTTCTTGTGAGGATGGAAACCAATCTGATGATAAACCTATCTGACCCTGCTGTACAATTTTGCCTATCTTTCATTGCCATGAATGTAGCTGAGATCGGACTCAGGAGGGTGCAGGAATCTTGGAATTCGCATCCGATTGAAGAAAACAGCGCTCGAATTCCAGATATAGTTGCGCAAAGGATGTGTAGAGTACACCCAGTAAATCCATCTTTGGTTCCAAGTGTCGAGGATGTTATTCAATTGTACATTGACGCTGGAGGCACCATAACACTCAGTTCTTCATTTGGGGTAGATCCCCTGGCAAGCAGTGACGCACTCATCAATCGACGTGAAACAGAATTTACACAGAATAATCCTTCTTTCGACGATATTTACAGCAACGTTATAAATGGAGATGGGTCCTTTATGGCCGCTGCAGTCTTGTCATTTATTAACATCACTAGAAATTTGACACCTTGA	639	20	133	144
```

14. Calculate CpG O/E based on [Gavery and Roberts 2010](http://www.biomedcentral.com/1471-2164/11/483).

```
$ awk '{print $1, "\t", (($4)/($5*$6))*(($3^2)/($3-1))}' comb > Mcap_ID_CpG
## use ^ instead of ** for exponent

Output:
awk: division by zero
 input record number 50330, file comb
 source line number 1
```

15. Checking CpG O/E calculations for first 10 genes in file.

```
$ head Mcap_ID_CpG

Output:
adi2mcaRNA10013_R0 	 0.555527
adi2mcaRNA10016_R5 	 0.751126
adi2mcaRNA10017_R1 	 0.751626
adi2mcaRNA10018_R0 	 0.767884
adi2mcaRNA10022_R0 	 0.861441
adi2mcaRNA10025_R0 	 0.647868
adi2mcaRNA10028_R1 	 0.707412
adi2mcaRNA10028_R3 	 0.591402
adi2mcaRNA10043_R1 	 0.67171
adi2mcaRNA10065_R8 	 0.793626
```

**Annotating gene sets using Blast and GOSlim**

1.

```
blastx \
-query Mcap.mRNA.fa \ #FASTA file
-db ~blast/db/uniprot_sprot \ #Use your blastx database address
-max_target_seqs 1 \ #maximum number of target sequences = 1
-max_hsps 1 \ #maximum number of high-scoring pairs = 1
-outfmt 6 \ #output format = tabular
-evalue 1E-05 \ #E-value = 10^-5
-num_threads 8 \ #number of threads = 8
-out ../CpG_OE/Mcap_blastx_uniprot.tab \ #Direct output to designated directory
2> ../CpG_OE/Mcap_blastx_uniprot.error #Direct standard error output to its own file
```
