---
layout: post
title: Soft and Hard Homogenization Protocol
date: '2019-06-03'
categories: Protocols
tags: soft homogenization, hard homogenization, DNA extraction, RNA extraction, protocol
projects: Holobiont Integration
---

# Soft and Hard Homogenization Protocol    

Major Goal: To extract DNA and RNA from coral fragments and to enrich for host in the first "soft homogenization" fraction and holobiont in the "hard homogenization" fraction. Testing to see if "Soft homogenization" will include DNA and RNA from primarily the host for transcriptomics and DNA methylation analysis. "Hard homogenization" will include the DNA from the symbionts and microbiome for ITS2 and 16s amplification and Holobiont RNA.

Major Results: In the below protocol, the "soft" and "hard" homogenizations did not show a difference quantitatively or qualitatively in symbiont communities. "Mix S" = a mixture of soft homogenization extractions and "Mix H" = a mixture of hard homogenization extractions. There is no clear difference in extraction method.

![sym](https://github.com/emmastrand/EmmaStrand_Notebook/blob/master/images/sym-hard-soft.png?raw=true)

![sym2](https://github.com/emmastrand/EmmaStrand_Notebook/blob/master/images/sym-hard-soft-2.png?raw=true)

Major Take Home: Although this protocol is not effective, there is still a need to develop an efficient way to separate symbiont and host tissue for downstream genomic analysis. Ideas or suggestions? hputnam@uri.edu or emma_strand@uri.edu.

### Protocol Preparation

**Reagents and supplies**
- RNAse Away
- RNA/DNA shield   
- RNAse-free water
- Proteinase K digestion buffer
- Proteinase K  
- Isopropanol  
- 10mM Tris HCl pH 8.0 made with RNase-free water  
- 10% bleach
- Zymo Duet DNA/RNA Extraction kit buffers: DNA/RNA lysis buffer, DNA/RNA Prep Buffer, DNA/RNA Wash Buffer, DNA Digestion buffer, and DNase I

**Equipment**  
- Rocking oven that can be set at 55 &deg;C
- Thermomixer
- Vortexer  
- Qiagen Tissue Lyser: [Handbook](https://www.qiagen.com/us/resources/resourcedetail?id=65e7826c-4d50-4faf-8154-2fbc782c41a6&lang=en)
- Bead tubes: 2mL 0.5mm glass beads and tubes from [Fisher Scientific](https://www.fishersci.com/shop/products/bead-tube-2ml-0-5mm-glass-50pk/15340152)
- Designated RNAse free space  
- Tabletop and larger centrifuges for 1.5 mL capable of 12,000xg  
- Clippers

### Reagent Preparation

1. Add 96 mL 100% ethanol (104 mL 95% ethanol) to the 24 mL DNA/RNA Wash Buffer concentrate before use. DNA/RNA Wash Buffer included with D7003T (Mini Prep Plus Kit) is supplied ready-to-use and does not require the addition of ethanol prior to use. Check kit contents and instructions to confirm prep steps.  
2. Reconstitute the lyophilized (freeze-dried) DNase I as indicated on the vial prior to use. Mix by inversion. Store frozen aliquots.  
3. Reconstitute the lyophilized (freeze-dried) 20 mg Proteinase K with 1040 uL Proteinase K Storage Buffer or lyophilized (freeze-dried) 5 mg Proteinase K with 260 uL Proteinase K Storage Buffer. Vortex to dissolve. Store at -20 &deg;C.

### Fragment Preparation

Sterilizing working area:
I. Rinse clippers with:
  1. 10% bleach solution  
  2. DI water  
  3. Isopropanol  
  4. RNAse free water

II. Spray gloves with RNAse away and rub hands together.

Take fragments one at a time out of the freezer and sterilize clippers in between every fragment. These steps are time sensitive to prevent the coral fragments and freezer from thawing. Do not rush, but be efficient.

Assign "extraction numbers" to each ID for ease of labeling. See below chart for an example:  

| Extraction # | Coral ID | Species     | Homogenization |
|--------------|----------|-------------|----------------|
| 1            | 1123     | Montipora   | Soft           |
| 2            | 1123     | Montipora   | Hard           |
| 3            | 1769     | Montipora   | Soft           |
| 4            | 1769     | Montipora   | Hard           |
| 5            | 1056     | Pocillopora | Soft           |
| 6            | 1056     | Pocillopora | Hard           |
| 7            | 1607     | Pocillopora | Soft           |
| 8            | 1607     | Pocillopora | Hard           |

Clipping fragment:  
1. Label the lid of a bead tube with the coral fragment's sample ID number, and the side of the tube with the coral fragment ID and today's date.  
2. Add 1000 μl of RNA/DNA shield to the bead tube.
3. Remove desired fragment from -80 &deg;C freezer.  
4. Using sterilized clippers, clip off 1-2 small pieces and place into the bead tube. RNA/DNA shield needs to cover the fragments.

*Repeat above steps 1-5 for each fragment before moving on.*

### Soft and Hard Homogenization

5. Vortex for *Pocillopora* fragments for 1 minute and the *Montipora* fragments for 2 minutes. Leave the settings on and on max power. This is the "soft homogenization" step.  
6. After vortex time, check to see if the tissue is coming off of the skeleton and look for coloration change in shield liquid.  
7. Remove supernatant from the bead tube and place in a new 1.5 microcentrifuge tube labeled on the side with the extraction ID number, coral fragment's sample ID number, "soft", and today's date. Label the cap of the microcentrifuge tube with the extraction number. This supernatant will become the sample for "soft homogenization". The remaining chunks of fragment will become the sample for "hard homogenization".  
8. Add 500 μl of RNA/DNA shield to the bead tube with the coral fragment pieces. RNA/DNA shield needs to cover the fragment pieces.   
9. Place the beads tubes with the coral fragment pieces in the Qiagen Tissue Lyser Adapter Set 2 x 24.  
10. Operate the Tissue Lyser for 1 minute at 20 Hz.       
11. Remove the supernatant and put in a new 1.5 mL microcentrifuge tube labeled on the side with the extraction ID number, coral fragment's sample ID number, "hard", and today's date. Label the cap of the microcentrifuge tube with the extraction ID number.  
12. Add 300 μl of sample, 30 μl of Proteinase K digestion buffer (10:1 ratio of sample:digestion buffer), and 15 μl of Proteinase K (2:1 ratio of digestion buffer:Proteinase K) to a new 1.5 mL microcentrifuge tube.  
> At this point, each coral fragment should have 2 tubes to move forward to extractions with: one for soft homogenization and one for hard homogenization.  
For 4 coral fragments: 4 bead tubes with coral fragment pieces, 8 tubes with leftover homogenate sample (4 soft, 4 hard), and 8 tubes with homogenate, Proteinase K digestion buffer, and Proteinase K (4 soft, 4 hard).   

13. Vortex and spin down all tubes.   
14. Place bead tubes and leftover homogenate tubes in -80 &deg;C freezer.  


## Zymo Duet RNA DNA Extractions

Modified from the [Zymo protocol](https://github.com/kevinhwong1/KevinHWong_Notebook/blob/master/images/Zymo_quick-dna-rna_miniprep_plus_kit.pdf).

**DNA Extraction**  
1. Set up yellow DNA spin columns and collection tubes, label appropriately  
2. Warm elution liquids to 70 &deg;C (10mM Tris HCl pH. 8.0 and RNase free water)  
3. Add equal volume (to supernatant; 345 µl) DNA/RNA lysis buffer to each sample tube  
4. Finger flick to mix tubes  
5. Add 700 µl (total volume) of sample gently to the yellow DNA spin column  
6. Centrifuge at 16,000 rcf (g) for 30 seconds  
7. **Important** Save the flow through from this step: transfer to a new 1.5mL tube labeled for RNA  
8. Add 400 µl DNA/RNA Prep Buffer gently to the yellow DNA spin columns  
9. Centrifuge at 16,000 rcf (g) for 30 seconds  
10. Discard flow through (Zymo kit waste)  
11. Add 700 µl DNA/RNA Wash Buffer gently to the yellow DNA spin columns  
12. Centrifuge at 16,000 rcf (g) for 30 seconds  
13. Discard flow through (Zymo kit waste)  
14. Add 400 µl DNA/RNA Wash Buffer genetly to the yellow DNA spin columns  
15. Centrifuge at 16,000 rcf (g) for **2 minutes**  
16. Discard flow through (Zymo kit waste)  
17. Transfer yellow columns to new 1.5 mL microcentrifuge tubes ("E1")
18. Add 10 µl warmed 10mM Tris HCl to each yellow DNA column by dripping slowly directly on the filter    
19. No incubation for the first elution. Centrifuge at 16,000 rcf (g) for 30 seconds  
20. Transfer the yellow DNA columns to new 1.5 mL microcentrifuge tubes ("E2")   
21. Add 100 µl warmed 10mM Tris HCl to each yellow DNA column by dripping slowly directly on the filter   
22. Incubate at room temperature for 15 minutes.   
23. Centrifuge at 16,000 rcf (g) for 30 seconds  
24. Aliquot 10 µl of the second elution ("E2") to 0.5 mL PCR tubes for Qubit and Gel Electrophoresis analysis.    
25. Store at 4 &deg;C if quantifying the same day or the next, if waiting longer store in -20 &deg;C freezer

**RNA Extraction**
*Can do concurrently with DNA Extraction after DNA Extraction Step 7*  
1. Add equal volume (700 µl) 100% EtOH to the 1.5 mL tubes labeled for RNA containing the original yellow column flow through  
2. Vortex and spin down to mix  
3. Add 700 µl of that liquid to the green RNA spin columns  
4. Centrifuge at 16,000 rcf (g) for 30 seconds  
5. Discard flow through (Zymo kit waste)  
6. Add 700 µl to the green RNA spin columns (the rest from the 1.5mL RNA tubes)  
7. Centrifuge at 16,000 rcf (g) for 30 seconds
    - Get DNase I from freezer  
8. Discard flow through (Zymo kit waste)  
9. Add 400 µl DNA/RNA Wash Buffer gently to each green RNA column  
10. Centrifuge at 16,000 rcf (g) for 30 seconds  
11. Discard flow through (Zymo kit waste)  
12. Make DNase I treatment master mix:
    - 75µl DNA Digestion buffer x # of samples
    - 5µl DNase I x # of samples  
> 4 coral fragments (8 samples total): 600 µl of buffer and 40 µl of DNase I  
8 coral fragments (16 samples total): 1,200 µl of buffer, 80 µl of DNase I  
10 coral fragments (20 samples total): 1,500 µl of buffer, 100 µl of DNase I. Place in 5 mL tube instead of 1.5 mL tube.

13. Add 80 µl DNase I treatment master mix directly to the filter of the green RNA columns  
14. Incubate at room temp for 15 minutes  
15. Add 400 µl DNA/RNA Prep Buffer gently to each column  
16. Centrifuge at 16,000 rcf (g) for 30 seconds  
17. Discard flow through (Zymo kit waste)  
18. Add 700 µl DNA/RNA Wash Buffer gently to the green RNA spin columns  
19. Centrifuge at 16,000 rcf (g) for 30 seconds  
20. Discard flow through (Zymo kit waste)  
21. Add 400 µl DNA/RNA Wash Buffer genetly to the green RNA spin columns  
22. Centrifuge at 16,000 rcf (g) for **2 minutes**  
23. Discard flow through (Zymo kit waste)  
24. Transfer green columns to new 1.5 mL microcentrifuge tubes  
25. Add 50 µl warmed DNase/RNase free water to each green RNA column by dripping slowly directly on the filter  
26. Incubate at room temp for 5 minutes  
27. Centrifuge at 16,000 rcf (g) for 30 seconds  
28. Repeat steps 25-27 for a final elution volume of 100 µl  
29. Aliquot 5 µl of the final elution to 0.5 mL PCR tubes for Qubit and TapeStation analysis.    
30. Store all tubes in the -80 &deg;C freezer.

## Clean-up
1. Place tissue and liquid in the waste container labeled Zymo extraction waste.  
2. Wipe down RNA free area with RNase away and kimwipes.  
3. Throw away all tips and restock tip boxes if necessary.

## Testing Quantity and Quality
To test RNA and DNA quantity: [Qubit](https://github.com/emmastrand/EmmaStrand_Notebook/blob/master/_posts/2019-05-31-Qubit-Protocol.md)  
To test RNA quality: [TapeStation](https://github.com/emmastrand/EmmaStrand_Notebook/blob/master/_posts/2019-05-31-TapeStation-Protocol.md)  
To test DNA quality: [Gel Electrophoresis](https://github.com/emmastrand/EmmaStrand_Notebook/blob/master/_posts/2019-07-16-Gel-Electrophoresis-Protocol.md)

### Solution Key:
**DNA/RNA Lysis Buffer**: Contains a high concentration of chaotropic salts, which destabilize hydrogen bonds, van der Waals forces and hydrophobic interactions. This destabilizes proteins, including nucleases. The buffer will also disrupt association of nucleic acids with water.    
**DNA/RNA Wash Buffer**: Used to remove residual proteins and carbohydrates, and purifies the molecular end product.  
**DNase I**: deoxyribonuclease is an enzyme that catalyzes the hydrolytic cleavage of phosphodiester linkages in the DNA backbone, thus degrading DNA. This is used in the RNA extraction step to purify RNA final product.  
**DNA/RNA Prep Buffer**: Ethanol and guanidinium chloride used a deproteinization measure. Guanidinium is a very strong denaturing agent that will dissolve cells and RNAses.  
**EtOH**: the addition of alcohol further enhances and influences the binding of nucleic aicds to the silica (see spin colum description below).  
**10 mM Tris HCl**: DNA is more stable at slightly basic pH and will dissolve faster in a buffer than water. Using Tris instead of water will enhance rehydration of high molecular weight DNA.  
**DNA/RNase free water**: Used as an elution in the RNA extraction steps. RNA can tolerate a slightly acidic pH and dissolves. Warmed water as an elution increases high molecular yield.  
**Yellow DNA/Green RNA Spin Columns**: contain a silica membrane that binds to nucleic acids. Proteins and polysaccharides should be in the flow-through liquid that is discarded while RNA and DNA is kept on the membrane.    
**Proteinase K**: an enzyme that cleaves the peptide bond in proteins next to the carboxyl group of hydrophobic amino acid residues. This decontaminates the sample and leaves purified DNA/RNA.

General workflow of an extraction:  
1. Cell lysis through disruption of cellular membranes  
2. Dehydration and precipitation of the cellular proteins (protein denaturation)  
3. Separation of cellular proteins and other cellular components out of the nucleic acid  
4. Precipitation and dissolving the nucleic acid  

Helpful links:  
[How DNA RNA extraction kits work](https://bitesizebio.com/13516/how-dna-extraction-rna-miniprep-kits-work/)  
[DNA Extraction and Purification](https://www.labome.com/method/DNA-Extraction-and-Purification.html)  
[DNA Extraction Methods](https://www.sciencedirect.com/topics/neuroscience/dna-extraction)

### Troubleshooting  

More info coming soon.  
