---
layout: post
title: M.capitata Genome v3 Functional Annotation
date: '2022-11-02'
categories: Analysis
tags: genome, bioinformatics
projects: KBay Bleaching Pairs
---

# Montipora capitata v3 Functional Genome Annotation

Contents:  
- [**Setting Up Andromeda**](#Setting_up) 
- [**Align query protein sequences against databases**](#blast)  

### References:  

For detailed explanations of each step, refer to the below pipelines. 

**Functional Annotation** 

- Erin Chille Functional Annotation: https://github.com/echille/Mcapitata_Developmental_Gene_Expression_Timeseries/blob/master/0-BLAST-GO-KO/2020-10-08-M-capitata-functional-annotation-pipeline.md  
- Jill Ashey Functional Annotation: https://github.com/JillAshey/FunctionalAnnotation/blob/main/FunctionalAnnotation_Worflow.md  
- Kevin Wong Functional Annotation: https://github.com/hputnam/Past_Genome/blob/master/genome_annotation_pipeline.md#functional-annotation-1
- Danielle Becker-Polinski Functional Annotation: https://github.com/daniellembecker/DanielleBecker_Lab_Notebook/blob/master/_posts/2021-12-08-Molecular-Underpinnings-Functional-Annotation-Pipeline.md#overview


## <a name="Setting_up"></a> **Setting Up Andromeda** 

### Shared Protein Databases from the Putnam Lab

Make directory for genomes: `mkdir genome_annotation` & `mkdir genome_annotation/scripts`

Trembl path: `/data/putnamlab/shared/databases/trembl_db/uniprot_trembl.fasta`  
Swiss: `/data/putnamlab/shared/databases/swiss_db/uniprot_sprot.fasta`
Uniprot: `/data/putnamlab/shared/databases/nr/`. 

Download the protein fasta files from genome: 

`wget http://cyanophora.rutgers.edu/Pocillopora_acuta/Pocillopora_acuta_HIv2.genes.pep.faa.gz` & `gunzip Pocillopora_acuta_HIv2.genes.pep.faa.gz` (This one will be used for another genome annotation)

`wget http://cyanophora.rutgers.edu/montipora/Montipora_capitata_HIv3.genes.pep.faa.gz` & `gunzip Montipora_capitata_HIv3.genes.pep.faa.gz`. (This will be used in the below scripts). 

**Count number of protein sequences.** 

```
zgrep -c ">" /data/putnamlab/estrand/Montipora_capitata_HIv3.genes.pep.faa 

output: 54384
```

## <a name="blast"></a> **Align query protein sequences against databases** 

### NCBI nr database

Output: .tab and .xml file of aligned sequence info. The .xml file is the important one - it will be used as input for Blast2GO

`mcap_blast_nr.sh`: 

```
#!/bin/bash
#SBATCH --job-name="MC-nr"
#SBATCH -t 240:00:00
#SBATCH --nodes=1 --ntasks-per-node=1
#SBATCH --mem=128GB
#SBATCH --account=putnamlab
#SBATCH --export=NONE
#SBATCH -D /data/putnamlab/estrand/genome_annotation #### this should be your new output directory so all the outputs ends up here
#SBATCH --error=../"%x_error.%j" #if your job fails, the error report will be put in this file
#SBATCH --output=../"%x_output.%j" #once your job is completed, any final job report comments will be put in this file

# load modules needed (specific need for my computer)
source /usr/share/Modules/init/sh # load the module function

# load modules needed
echo "START" $(date)
module load DIAMOND/2.0.0-GCC-8.3.0 #Load DIAMOND

echo "Updating Mcap annotation" $(date)
diamond blastp -b 2 -d /data/putnamlab/shared/databases/nr -q /data/putnamlab/estrand/Montipora_capitata_HIv3.genes.pep.faa -o Mcap_v3_blastp_annot -f 100 -e 0.00001 -k 1 --threads $SLURM_CPUS_ON_NODE --tmpdir /tmp/

echo "Search complete... converting format to XML and tab"

diamond view -a Mcap_v3_blastp_annot.daa -o Mcap_v3_blastp_annot.xml -f 5
diamond view -a Mcap_v3_blastp_annot.daa -o Mcap_v3_blastp_annot.tab -f 6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore qlen slen

echo "STOP" $(date)
```

Check how many hits this yielded:

```
$ wc -l Mcap_v3_blastp_annot.tab

output: 30990 Mcap_v3_blastp_annot.tab # ~X hits out of 54384
```

### SwissProt Database 

Output: .xml file of aligned sequence info

`mcap_swissprot_blast.sh`:

```
#!/bin/bash
#SBATCH --job-name="MC-swiss"
#SBATCH -t 240:00:00
#SBATCH --nodes=1 --ntasks-per-node=1
#SBATCH --mem=128GB
#SBATCH --account=putnamlab
#SBATCH --export=NONE
#SBATCH -D /data/putnamlab/estrand/genome_annotation #### this should be your new output directory so all the outputs ends up here
#SBATCH --error=../"%x_error.%j" #if your job fails, the error report will be put in this file
#SBATCH --output=../"%x_output.%j" #once your job is completed, any final job report comments will be put in this file

# load modules needed (specific need for my computer)
source /usr/share/Modules/init/sh # load the module function

# load modules needed
echo "START" $(date)
module load BLAST+/2.11.0-gompi-2020b #load blast module

echo "Blast against swissprot database" $(date)
blastp -max_target_seqs 5 -num_threads 20 -db /data/putnamlab/shared/databases/swiss_db/swissprot_20220307 -query /data/putnamlab/estrand/Montipora_capitata_HIv3.genes.pep.faa -evalue 1e-5 -outfmt '5 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore qlen' -out Mcap_v3_swissprot_protein.out

echo "STOP" $(date)
```

### Trembl database 